trajectory_tracker/test/include/trajectory_tracker_test.h

```diff
@@ -115,66 +114,70 @@ class TrajectoryTrackerTest : public ::testing::Test
+  TrajectoryTrackerTest(const std::string& node_name = "test_trajectory_tracker")
+  : delay_(rclcpp::Duration(0, 0))
   {
-    sub_cmd_vel_ = nh_.subscribe(
-        "cmd_vel", 1, &TrajectoryTrackerTest::cbCmdVel, this);
-    sub_status_ = nh_.subscribe(
-        "trajectory_tracker/status", 1, &TrajectoryTrackerTest::cbStatus, this);
-    pub_path_ = nh_.advertise<nav_msgs::Path>("path", 1, true);
-    pub_path_vel_ = nh_.advertise<trajectory_tracker_msgs::PathWithVelocity>("path_velocity", 1, true);
-    pub_odom_ = nh_.advertise<nav_msgs::Odometry>("odom", 10, true);
+    node_ = rclcpp::Node::make_shared(node_name);
+    using std::placeholders::_1;
+    sub_cmd_vel_ = node_->create_subscription<geometry_msgs::msg::Twist>(
+      "cmd_vel", 1, std::bind(&TrajectoryTrackerTest::cbCmdVel, this, _1));
+    sub_status_ = node_->create_subscription<trajectory_tracker_msgs::msg::TrajectoryTrackerStatus>(
+      "trajectory_tracker/status", 1, std::bind(&TrajectoryTrackerTest::cbStatus, this, _1));
+    pub_path_ = node_->create_publisher<nav_msgs::msg::Path>("path", rclcpp::QoS(1).transient_local());
+    pub_path_vel_ = node_->create_publisher<trajectory_tracker_msgs::msg::PathWithVelocity>(
+        "path_velocity", rclcpp::QoS(1).transient_local());
+    pub_odom_ = node_->create_publisher<nav_msgs::msg::Odometry>("odom", rclcpp::QoS(10).transient_local());
 
     double delay;
-    pnh_.param("odom_delay", delay, 0.0);
-    delay_ = ros::Duration(delay);
-    pnh_.param("error_lin", error_lin_, 0.01);
-    pnh_.param("error_large_lin", error_large_lin_, 0.1);
-    pnh_.param("error_ang", error_ang_, 0.01);
+    delay = node_->declare_parameter<double>("odom_delay", 0.0);
+    delay_ = rclcpp::Duration::from_seconds(delay);
+    error_lin_ = node_->declare_parameter<double>("error_lin", 0.01);
+    error_large_lin_ = node_->declare_parameter<double>("error_large_lin", 0.1);
+    error_ang_ = node_->declare_parameter<double>("error_ang", 0.01);
 
-    dynamic_reconfigure_client_.reset(new dynamic_reconfigure::Client<ParamType>("/trajectory_tracker"));
+    tfb_ = std::make_unique<tf2_ros::TransformBroadcaster>(node_.get());
 
-    ros::Rate wait(10);
+    rclcpp::Rate wait(10);
     for (size_t i = 0; i < 100; ++i)
     {
       wait.sleep();
-      ros::spinOnce();
-      if (pub_path_.getNumSubscribers() > 0)
+      rclcpp::spin_some(node_);
+      if (pub_path_->get_subscription_count() > 0)
         break;
     }
   }
   void initState(const tf2::Transform& pose)
   {
     // Wait trajectory_tracker node
-    ros::Rate rate(10);
-    const auto start = ros::WallTime::now();
-    while (ros::ok())
+    rclcpp::Rate rate(10);
+    rclcpp::Clock clock;
+    const auto start = clock.now();
+    while (rclcpp::ok())
     {
-      nav_msgs::Path path;
+      nav_msgs::msg::Path path;
       path.header.frame_id = "odom";
-      path.header.stamp = ros::Time::now();
-      pub_path_.publish(path);
+      path.header.stamp = clock.now();
```

Comment:
It is better choice to use node clock for ROS messages.
```suggestion
      path.header.stamp = node_->now();
```